dropwizard-configuration/src/main/java/io/dropwizard/configuration/YamlConfigurationFactory.java

```diff
@@ -105,9 +105,9 @@ public T build(ConfigurationSourceProvider provider, String path) throws IOExcep
     @Override
     public T build() throws IOException, ConfigurationException {
         try {
-            final JsonNode node = mapper.valueToTree(klass.getConstructor().newInstance());
+            final JsonNode node = mapper.valueToTree(klass.newInstance());
```

Comment:
Mind updating the commit message? I don't think it's clear what this change is addressing, exactly.